.github/workflows/rust.yml

```
@@ -23,3 +23,5 @@ jobs:
       run: cargo build --verbose
     - name: Run tests
       run: cargo test --verbose
+    - name: Check formatting
+      run: cargo fmt --all -- --check
```

I propose to move this before cargo build because the check is fast, and it's better for the CI job to fail fast in case formatting is incorrect, so we know we have to fix the formatting and we don't need to wait for the full compilation.

---

I thought it would be a good idea for the developer to verify the logic before handling the formatting, but I see your point about not having to wait for all the tests to pass before checking if the format is correct
Will change